### DISCUSSION TITLE
NL dimension in ScalePyramid

### DISCUSSION THREAD

**Discussion by vinod1234567890** [kornia/kornia#829] [Questions] [answered]
I want to use a scale pyramid to extract resize invariant features for an image similarity project. I'm using resnet18 trained with triplet loss. I want to extract features at multiple scales and average pool them. 

But when I pass the image tensor through kornia.geometry.transform ScalePyramid, it is returning images with shape B,C,NL,H,W. 

I'm not sure how to understand the NL dimension. I need tensors in the shape B,C,H,W. How can I drop this NL dimension? If I should not drop it, do I need to make use of it?

**Comment by ducha-aiki** [comment, marked as answer]
Answered here, putting the link for convenience
https://discuss.pytorch.org/t/nl-dimension-in-kornias-scalepyramid/107413
